### PR TITLE
add icons for Gas, Power, Problem and Safety

### PR DESCRIPTION
**Pull request by Mariusthvdb**
to allow for more fine-grained icon representation in the frontend.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by kukulich** [comment]
You also changed icons for “gas”.

**Comment by Mariusthvdb** [comment]
> You also changed icons for “gas”.

didn't realize that, doing so by leaving them open... Now added alert biohazard for Gas, and reversed the on/off icons, which I had mixed up.

File: src/common/entity/binary_sensor_icon.ts [hunk 1]
```diff
@@ -16,9 +16,13 @@ export const binarySensorIcon = (state: HassEntity) => {
     case "garage_door":
       return activated ? "hass:garage" : "hass:garage-open";
     case "gas":
+      return activated ? "hass:shield-check" : "hass:biohazard";
     case "power":
+      return activated ? "hass:power-off" : "hass:power";
     case "problem":
+      return activated ? "hass:check-circle" : "hass:alert-circle";
     case "safety":
+      return activated ? "hass:check-circle" : "hass:alert-circle";
```
Comment (bramkragten) [inline src/common/entity/binary_sensor_icon.ts:25]: The icon for smoke looks more fitting for safety?

File: src/common/entity/binary_sensor_icon.ts [hunk 2]
```diff
@@ -16,9 +16,13 @@ export const binarySensorIcon = (state: HassEntity) => {
     case "garage_door":
       return activated ? "hass:garage" : "hass:garage-open";
     case "gas":
+      return activated ? "hass:shield-check" : "hass:biohazard";
```
Comment (bramkragten) [inline src/common/entity/binary_sensor_icon.ts:19]: gas really isn't a biohazard, is it? We could just leave it how it was.

File: src/common/entity/binary_sensor_icon.ts (hunk 1 above)
Comment (Mariusthvdb) [reply src/common/entity/binary_sensor_icon.ts:25]: Id like that for Gas...

maybe simply use

```
return activated ? "hass:shield-check" : "hass:shield-alert";
```

for safety...

File: src/common/entity/binary_sensor_icon.ts (hunk 2 above)
Comment (Mariusthvdb) [reply src/common/entity/binary_sensor_icon.ts:19]: thought I had done that, but @kukulich said I changed the icon...
what about mdi:smoke-detector?
```
return activated ? "hass:shield-check" : "hass:smoke-detector";
```

Comment (bramkragten) [reply src/common/entity/binary_sensor_icon.ts:19]: If we don't have something specific lets just fall back to the default like smoke?

Comment (Mariusthvdb) [reply src/common/entity/binary_sensor_icon.ts:19]: right, entering:
```
      return activated ? "hass:shield-check" : "hass:alert";
```

File: src/common/entity/binary_sensor_icon.ts [hunk 3]
```diff
@@ -16,9 +16,13 @@ export const binarySensorIcon = (state: HassEntity) => {
...
     case "gas":
+      return activated ? "hass:shield-check" : "hass:alert";
     case "power":
+      return activated ? "hass:power-off" : "hass:power";
     case "problem":
+      return activated ? "hass:check-circle" : "hass:alert-circle";
     case "safety":
+      return activated ? "hass:check-circle" : "hass:alert-circle";
     case "smoke":
       return activated ? "hass:shield-check" : "hass:alert";
```
Comment (bramkragten) [inline src/common/entity/binary_sensor_icon.ts:27]:
```suggestion
    case "power":
      return activated ? "hass:power-off" : "hass:power";
    case "problem":
      return activated ? "hass:check-circle" : "hass:alert-circle";
    case "safety":
    case "gas":
    case "smoke":
      return activated ? "hass:shield-check" : "hass:shield-alert";
```
Maybe `fire-alert`, but fire <> smoke?

**Comment by bramkragten** [comment]
Please rebase this.

**Comment by bramkragten** [comment]
Any update on this?

**Pull request closed without merging**